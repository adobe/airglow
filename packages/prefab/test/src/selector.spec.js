/*
Copyright 2019 Adobe. All rights reserved.
This file is licensed to you under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License. You may obtain a copy
of the License at http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under
the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License.
*/

import { assocPath } from 'ramda';
import selector from '../../src/selector';


let crayonSelector;
let formSelector;
let validState;
const dispatch = 'Dispatch!';

const invalidState = {
  prefab: {
    crayonForm: {
      construct: {
        fields: 'config fields'
      },
      store: {
        fields: 'store fields'
      }
    }
  }
};

const spawnState = () => {
  formSelector = sinon.stub().returns({ data: 'results!' });
  validState = assocPath(
    ['prefab', 'crayonForm', 'construct'],
    { fields: 'config fields' },
    invalidState
  );
};

describe('prefab selector', () => {
  beforeEach(() => {
    spawnState();
    crayonSelector = selector(formSelector, 'crayonForm')('selector options!');
  });

  it('should properly pass the values into the selector', function () {
    crayonSelector(validState, dispatch);
    expect(formSelector.lastCall.args[0]).toMatchSnapshot();
  });
  it('should return the results of the selector', function () {
    expect(crayonSelector(validState, dispatch)).toMatchSnapshot();
  });
});
