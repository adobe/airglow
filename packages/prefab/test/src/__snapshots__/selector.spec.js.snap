// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prefab selector should properly pass the values into the selector 1`] = `
Object {
  "dispatch": "Dispatch!",
  "globalState": Object {
    "prefab": Object {
      "crayonForm": Object {
        "construct": Object {
          "fields": "config fields",
        },
        "store": Object {
          "fields": "store fields",
        },
      },
    },
  },
  "options": "selector options!",
  "props": undefined,
  "state": Object {
    "construct": Object {
      "fields": "config fields",
    },
    "store": Object {
      "fields": "store fields",
    },
  },
}
`;

exports[`prefab selector should return the results of the selector 1`] = `
Object {
  "data": "results!",
}
`;
